ios/Gekidou/Sources/Gekidou/Networking/ShareExtension+Delegate.swift

```diff
@@ -176,6 +184,14 @@ extension ShareExtension: URLSessionDataDelegate {
         
         let total = data.totalFiles
         let count = data.fileIds.count
+        if let timeout = data.failTimeout, timeout < Date() {
+            backgroundSession?.invalidateAndCancel()
+            os_log(
+                "Mattermost BackgroundSession: postMessageIfUploadFinished session %{public}@ timed out",
+                session.configuration.identifier ?? "no identifier"
+            )
+            return
+        }
```

Comment:
shouldn't this be done after checking if the last file was actually uploaded? because if that is the case, this is cancelling the whole thing instead of just posting the message

Reply:
Yup! The reason is that we already told the user that the sharing failed due to a timeout. Since we can't control when this happens this could be a day later making things confusing. So we cancel the first opportunity we have